src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        int bucketNumber = getBucketNumber(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        Node<K, V> currentNode = table[bucketNumber];
+        if (table[bucketNumber] == null) {
+            table[bucketNumber] = newNode;
+            size++;
+        } else {
+            while (currentNode != null) {
+                if (currentNode.key == null && key == null
+                        || currentNode.key != null && currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            newNode.next = table[bucketNumber].next;
+            table[bucketNumber].next = newNode;
+            size++;
+        }
+    }
 
+    private void resize() {
```

Comment:
Also, segregate this method into two methods cause you make resize mot every time when you call the method:
1. checkThreshold() - making check if you need to make resize;
2. resize() - contains logic only for resizing